test/extended/router/externalCertificate.go

```diff
@@ -0,0 +1,682 @@
+			_, err = oc.KubeClient().CoreV1().Secrets(oc.Namespace()).Create(context.Background(), secret, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Providing router service account permissions to get,list,watch the secret")
+			_, err = oc.KubeClient().RbacV1().Roles(oc.Namespace()).Create(context.Background(),
+				createRole(oc.Namespace(), "my-tls-secret"), metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			_, err = oc.KubeClient().RbacV1().RoleBindings(oc.Namespace()).Create(context.Background(),
+				createRoleBinding(oc.Namespace()), metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Creating multiple routes referencing same external certificate")
+			for i := 0; i < N; i++ {
+				route := createRouteWithEC(oc.Namespace(), routeNames[i], secret.Name, helloPodSvc, hosts[i], routev1.TLSTerminationEdge)
+				_, err = oc.RouteClient().RouteV1().Routes(oc.Namespace()).Create(context.Background(), route, metav1.CreateOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				routes = append(routes, route)
+			}
+		})
+
+		g.Describe("the router should support external certificate", func() {
+			g.It("and routes are reachable", func() {
+				g.By("Sending https request")
+				for i := 0; i < len(routes); i++ {
+					hostName, err := getHostnameForRoute(oc, routes[i].Name)
+					o.Expect(err).NotTo(o.HaveOccurred())
+					resp, err := httpsGetCall(hostName, rootDerBytes)
+					o.Expect(err).NotTo(o.HaveOccurred())
+					o.Expect(resp).Should(o.ContainSubstring("Hello OpenShift"))
+				}
+			})
+
+			g.Context("and the secret is deleted", func() {
+				g.BeforeEach(func() {
+					g.By("Deleting the secret")
+					err = oc.KubeClient().CoreV1().Secrets(oc.Namespace()).Delete(context.Background(), secret.Name, metav1.DeleteOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+				})
+
+				g.It("then routes are not reachable", func() {
+					g.By("Checking the route status")
+					for i := 0; i < len(routes); i++ {
+						checkRouteStatus(oc, routes[i].Name, corev1.ConditionFalse, "ExternalCertificateValidationFailed")
+					}
+				})
+
+				g.Context("and re-created again", func() {
+					g.BeforeEach(func() {
+						g.By("Re-creating the deleted secret")
+						_, err = oc.KubeClient().CoreV1().Secrets(oc.Namespace()).Create(context.Background(), secret, metav1.CreateOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+					})
+
+					g.It("then routes are reachable", func() {
+						g.By("Sending https request")
+						for i := 0; i < len(routes); i++ {
+							hostName, err := getHostnameForRoute(oc, routes[i].Name)
+							o.Expect(err).NotTo(o.HaveOccurred())
+							resp, err := httpsGetCall(hostName, rootDerBytes)
+							o.Expect(err).NotTo(o.HaveOccurred())
+							o.Expect(resp).Should(o.ContainSubstring("Hello OpenShift"))
+						}
+					})
+				})
+				g.Context("and re-created again but RBAC permissions are dropped", func() {
```

Comment:
dint get it, why "dropping the RBAC" is under  "**valid setup**"?

Reply:
The **"valid setup"**  establishes the preconditions required for a route with externalCertificate to function correctly. The subsequent tests, including those involving dropping RBAC, build upon this valid setup to test how the route behaves under different modification scenarios (secret updates, deletions, re-creation or route changes). Without the initial valid configuration, these specific tests wouldn't be meaningful.

Dropping RBAC in this context is specifically designed to re-test the validation after the initial valid setup. It ensures the route behaves as expected (becomes unavailable) when permissions are dropped in between.